### PR TITLE
fix: Search for Hyphens

### LINKED ISSUE #2798: 🐛 [Bug] Searches with a hyphen return no results
> [sushidave](https://github.com/sushidave) **Authored by [sushidave](https://github.com/sushidave)**
_<time datetime="2020-01-16T10:35:22Z" title="Thursday, January 16th 2020, 11:35:22 am +01:00">Jan 16, 2020</time>_
_Closed <time datetime="2020-03-18T20:40:10Z" title="Wednesday, March 18th 2020, 9:40:10 pm +01:00">Mar 18, 2020</time>_
---

## :bug: Bugreport

A search that contains a hyphen returns no results, neither posts nor users.

Example 1:

hc-bug-search-doesnt-return-dash-example-1a

Example 2:
hc-bug-search-doesnt-return-dash-example-1b

Example 3:
hc-bug-search-doesnt-return-dash-example-2

### Steps to reproduce the behavior
1. Type in a string containing a hyphen in the search field.
2. The search results show "Nothing found".

### Expected behavior

The query returns correct results.

### Version & Environment
 Type: [any]      
 - OS: [any]      
 - Browser: [any] 
 - Version [any]  
 - Device: [any]

### PR DESCRIPTION
**Pull request by Tirokk**
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-02-06T20:39:49Z" title="Thursday, February 6th 2020, 9:39:49 pm +01:00">Feb 6, 2020</time>_
_Closed <time datetime="2020-03-10T00:13:39Z" title="Tuesday, March 10th 2020, 1:13:39 am +01:00">Mar 10, 2020</time>_
---

## 🍰 Pullrequest
In the search query, the hyphen is replaced by a space.

### Issues
- fixes #2798

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-02-06T21:12:03Z" title="Thursday, February 6th 2020, 10:12:03 pm +01:00">Feb 6, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4931/specs"></a> <b title="Passed">59</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4931/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 0ba5aa3ddc |
| **Started** | Feb 6, 2020 8:59 PM |
| **Ended** | Feb 6, 2020 9:16 PM |
| **Duration** | 16:49 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Electron 78  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/4931/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: backend/src/schema/resolvers/searches.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ export default {
       // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
       const myQuery = query
         .replace(/\s+/g, ' ')
-        .replace(/[[@#:*~\\$|^\]?/"'(){}+?!,.-;]/g, '')
+        .replace(/[[@#:*~\\$|^\]?/"'(){}+?!,.;]/g, '')
+        .replace(/-/g, ' ')
```
Comment (Tirokk) [inline backend/src/schema/resolvers/searches.js:12]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-07T11:09:29Z" title="Friday, February 7th 2020, 12:09:29 pm +01:00">Feb 7, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -8,7 +8,8 @@ export default {
       // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
       const myQuery = query
         .replace(/\s+/g, ' ')
-        .replace(/[[@#:*~\\$|^\]?/"'(){}+?!,.-;]/g, '')
+        .replace(/[[@#:*~\\$|^\]?/"'(){}+?!,.;]/g, '')
+        .replace(/-/g, ' ')
```


This, as well, is the job of the tokenizer. Maybe you could search for "JavaScript tokenizer" to find a less hacky solution. Or look for a solution that is related to elastic search: "elastic search normalize query". 

You could add tests for this behaviour as well, at least for documentation purposes.

**Comment by Tirokk** [comment]
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-10T00:13:39Z" title="Tuesday, March 10th 2020, 1:13:39 am +01:00">Mar 10, 2020</time>_

---


I found a better solution.

**Review by Mogge — APPROVED** [review]
meh

Same feelings as with #3018 consider adding a test for it. I'll approve but not merge it.

**Pull request closed without merging**